apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs

```diff
@@ -101,11 +101,23 @@ public sealed class ReliableStreamReader : IDisposable
                 await ConnectAsync().ConfigureAwait(false);
 
                 Debug.Assert(_currentCall != null, "_currentCall != null");
-                cancellationToken.ThrowIfCancellationRequested();
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    _currentCall?.Dispose();
+                    _currentCall = null;
+                    cancellationToken.ThrowIfCancellationRequested();
+                }
 
                 for (int i = 0; i <= _resumeSkipCount; i++) {
                     //This calls a simple movenext on purpose.  If we get an error here, we'll fail out.
-                    _isReading = await _currentCall.ResponseStream.MoveNext(cancellationToken).WithSessionChecking(() => _session).ConfigureAwait(false);
+                    //TODO(benwu): pass in cancellationtoken and capture cancel to clean up _currentCall if needed.
```

Comment:
Is there an issue to track this?  If this is needed for GA I'd consider filing an issue.

Reply:
I don't believe we need to fix this for GA as its a limitation in (already GA'd) gRPC.  I'll let Jon comment on it however.

Reply:
I think we should at least note it as a limitation in the documentation somewhere - it's pretty surprising. I think I'd like to hold of on GA'ing in the current state until we at least know what we're going to do - I don't think we'll want it to be like this long-term.